public/app/features/dashboard/state/PanelQueryRunner.test.ts

```diff
@@ -1,3 +1,11 @@
+const applyFieldOverridesMock = jest.fn();
+
+jest.mock('@grafana/data', () => ({
+  __esModule: true,
+  ...(jest.requireActual('@grafana/data') as any),
+  applyFieldOverrides: applyFieldOverridesMock,
+}));
+
```

Comment:
Wouldn't 
```
jest.mock('@grafana/data', () => {
  return {
    applyFieldOverrides: applyFieldOverridesMock,
  }
});
```

work? Or do we need to keep some other functionality from `@grafana/data`?

Reply:
Without `requireActual` you end up with the following error:
```
 FAIL  public/app/features/dashboard/state/PanelQueryRunner.test.ts
  ● Test suite failed to run

    TypeError: Cannot read property 'Dark' of undefined

      89 | const darkTheme: GrafanaTheme = {
      90 |   ...defaultTheme,
    > 91 |   type: GrafanaThemeType.Dark,
         |                          ^
      92 |   isDark: true,
      93 |   isLight: false,
      94 |   name: 'Grafana Dark',

      at Object.<anonymous> (packages/grafana-ui/src/themes/dark.ts:91:26)
      at Object.<anonymous> (packages/grafana-ui/src/themes/getTheme.ts:1:1)
      at Object.<anonymous> (packages/grafana-ui/src/themes/ThemeContext.tsx:5:1)
      at Object.<anonymous> (packages/grafana-ui/src/components/Icon/Icon.tsx:5:1)
      at Object.<anonymous> (packages/grafana-ui/src/components/index.ts:1:1)
      at Object.<anonymous> (packages/grafana-ui/src/index.ts:6:1)
      at Object.<anonymous> (packages/grafana-runtime/src/config.ts:2:1)
      at Object.<anonymous> (packages/grafana-runtime/src/index.ts:7:1)
      at Object.<anonymous> (public/app/features/plugins/datasource_srv.ts:7:1)
      at Object.<anonymous> (public/app/features/dashboard/state/PanelQueryRunner.ts:7:1)
      at Object.<anonymous> (public/app/features/dashboard/state/PanelQueryRunner.test.ts:8:1)
```

Reply:
I see, makes sense, thanks for clarifying :)